kclvm/tools/src/query/override.rs

```diff
@@ -138,44 +326,182 @@ impl<'ctx> MutSelfMutWalker<'ctx> for OverrideTransformer {
+            let mut entry_paths = self.get_entry_paths(&mut entry.node);
+            paths.append(&mut entry_paths);
         }
-        (paths, paths_with_id)
+        paths
     }
-    pub(crate) fn _get_key_value_paths(
-        &mut self,
-        _entry: &mut ast::ConfigEntry,
-    ) -> (Vec<String>, Vec<String>) {
-        (vec![], vec![])
+
+    /// Get all field paths from a config entry.
+    fn get_entry_paths(&mut self, entry: &mut ast::ConfigEntry) -> Vec<String> {
+        let mut paths = vec![];
+        let path = self.get_path_from_key(&entry.key);
+        if !path.is_empty() {
+            paths.push(path.clone());
+            let value_paths = self.get_field_paths(&mut entry.value);
+            if !value_paths.is_empty() {
+                paths.append(
+                    &mut value_paths
+                        .iter()
+                        .map(|value_path| format!("{}.{}", path, value_path))
+                        .collect::<Vec<String>>(),
+                );
+            }
+        }
+        paths
+    }
+
+    /// Get config key path from the AST key node.
+    #[inline]
+    fn get_path_from_key(&mut self, key: &Option<ast::NodeRef<ast::Expr>>) -> String {
+        match key {
+            Some(key) => match &key.node {
+                ast::Expr::Identifier(identifier) => identifier.get_name(),
+                ast::Expr::StringLit(string_lit) => string_lit.value.clone(),
+                _ => "".to_string(),
+            },
+            None => "".to_string(),
+        }
     }
-    pub(crate) fn _find_schema_config_and_repalce(
+}
+
+/// OverrideNode is used to place different AST configuration syntax structures.
+#[derive(Debug)]
+enum OverrideNode<'a> {
+    Schema(&'a mut ast::SchemaExpr),
+    Config(&'a mut ast::ConfigExpr),
+}
+
+impl OverrideTransformer {
+    /// Lookup schema config all fields and replace if it is matched with the override spec,
+    /// return whether is found a replaced one.
+    fn lookup_schema_config_and_replace(&mut self, schema_expr: &mut ast::SchemaExpr) -> bool {
+        let paths = self.get_schema_config_field_paths(schema_expr);
+        match paths.iter().position(|r| r == &self.field_path) {
+            Some(pos) => {
+                let mut config = OverrideNode::Schema(schema_expr);
+                self.replace_node_with_path(&mut config, &paths[pos]);
+                true
+            }
+            None => false,
+        }
+    }
+
+    /// Replace AST node with path including schema_expr and config_expr.
+    fn replace_node_with_path(&mut self, config: &mut OverrideNode, path: &str) {
+        // Do not replace empty path parts on the config expression.
+        if path.is_empty() {
+            return;
+        }
+        // Split a path into multiple parts. `a.b.c` -> ["a", "b", "c"]
+        let parts = path.split('.').collect::<Vec<&str>>();
+        for i in 0..parts.len() {
+            match config {
```

Comment:
Simplify the match cases, try to replace with one replace_config_expr_with_path_parts function call.

Reply:
> Simplify the match cases, try to replace with one replace_config_expr_with_path_parts function call.

fixed.